### ISSUE TITLE
Vectors animations getting stuck on rapid zoom

### ISSUE THREAD

**Issue by mourner** [Leaflet/Leaflet#8101]
- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported (maybe #3249 is related?)

**Steps to reproduce**

Zoom any Leaflet map with vectors such as the [GeoJSON example](https://leafletjs.com/SlavaUkraini/examples/choropleth/example.html) rapidly back and forth

**Expected behavior**

Smooth animation at all points.

**Current behavior**

The animation often gets stuck, making the experience glitchy. This is a bug that has been annoying me for a long time on any Leaflet maps in the wild, but I have never investigated it properly, and it's a little hard to trigger and debug consistently. But would be amazing to find a fix.

https://user-images.githubusercontent.com/25395/161034664-28ce7fea-f87f-4b0f-ab39-377391f51934.mov

**Environment**
- Leaflet version: v1.7.1, v1.8.0-beta.1
- Browser (with version): Chrome v100
- OS/Platform (with version): macOS Monterey 12.3

**Additional context**

This seems to be limited to vector layers, but reproducible with both SVG and Canvas. Markers are not affected.

**Minimal example reproducing the issue**

https://leafletjs.com/SlavaUkraini/examples/choropleth/example.html

@IvanSanchez have you noticed this glitch? I'm wondering if we could poke at this a little to maybe get it fixed for the v1.8.0 final.

**Comment by IvanSanchez** [comment]
> have you noticed this glitch?

@mourner Nope, it's the first time I notice this. Right now I'm suspecting it's due to a `L.Renderer` calling `_update` in between a CSS zoom animation, i.e. between the `zoomstart` and `zoomend` events.

SVG renderers do an `_update()` on `moveend` but also on `zoomstart` (due to https://github.com/Leaflet/Leaflet/pull/3950). I wonder if there's some race condition between those events. Maybe some event logging will help?

>  I'm wondering if we could poke at this a little to maybe get it fixed for the v1.8.0 final.

Not sure. Looks like quite some work for a something that's not so easy to reproduce. I'm hoping for the post-1.8 removal of CSS animations anyway.

**Comment by mourner** [comment]
Also, it's reprodicible on Safari, but not Firefox, so looks like it might be triggering some Webkit/Blink CSS anim bug.

**Comment by IvanSanchez** [comment]
> Also, it's reprodicible on Safari, but not Firefox

I can reproduce it on Firefox 98 on Linux.

**Comment by Falke-Design** [comment]
I think a good sultion would be to check for now if it is IOS and for 1.8.1 removing it completley

**Comment by mourner** [comment]
@Falke-Design I'd like to include this in v1.8.0. Seems like a quick fix either way and we've localized it.

**Comment by mourner** [comment]
Poked at it on my iPhone for some time and I can't reproduce any issues after removing the hack apart from #7401. So let's just remove it for now and see if any issues pop up.

**Comment by IvanSanchez** [comment]
It's always nice to find some weird piece of code to armor against an obscure edge case, asking oneself "but why did we do this?"... and then being able to answer "Oh, *that*'s why".

I agree with removing the `animframe` and seeing if there's complaints from the iOS front.

**Comment by mourner** [comment]
Still happening in FF/Linus and occasionally Chrome/Mac, so will need further experiments.

**Comment by syonfox** [comment]
This is how I mitigate this issue on touch devices for mobile. It might help/ point in the right direction but it probably dosent fix the root bug. note that this must be initialized before the map so our event listener gets executed first

```
// https://github.com/Leaflet/Leaflet/issues/3249#issuecomment-1002020742
let mapDiv = document.getElementById('map')
L.DomEvent.on(mapDiv, 'wheel', (e) => {
    console.log('wheel')
    if (map.dragging.moving()) {
        map.dragging._onDragEnd(e);
    }
});

L.DomEvent.on(mapDiv, 'touchstart', (e) => {
    //https://github.com/Leaflet/Leaflet/blob/69f73ed1e8a83724c086ce206e7f9d7babc5d3dc/src/map/handler/Map.TouchZoom.js#L42
    if (!e.touches || e.touches.length !== 2 || map._animatingZoom /*|| this._zooming*/) {
        return;
    }
    console.log('touchstart')

    if (map.dragging.moving()) {
        map.dragging._onDragEnd(e);
    }
});
```
